src/ExtEvLoop.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+namespace React\EventLoop;
+
+use Ev;
+use EvLoop;
+use React\EventLoop\Tick\FutureTickQueue;
+use React\EventLoop\Timer\Timer;
+use SplObjectStorage;
+
+/**
+ * An `ext-ev` based event loop.
+ *
+ * This loop uses the [`ev` PECL extension](https://pecl.php.net/package/ev),
+ * that provides an interface to `libev` library.
+ *
+ * This loop is known to work with PHP 5.4 through PHP 7+.
+ *
+ * @see http://php.net/manual/en/book.ev.php
+ * @see https://bitbucket.org/osmanov/pecl-ev/overview
+ */
+class ExtEvLoop implements LoopInterface
+{
+    private $loop;
+    private $futureTickQueue;
+    private $timers;
+    private $readStreams = [];
+    private $writeStreams = [];
+    private $running;
+    private $signals;
+    private $signalEvents = [];
+
+    public function __construct()
+    {
+        $this->loop = new EvLoop();
+        $this->futureTickQueue = new FutureTickQueue();
+        $this->timers = new SplObjectStorage();
+        $this->signals = new SignalsHandler(
+            $this,
+            function ($signal) {
+                $this->signalEvents[$signal] = $this->loop->signal($signal, function () use ($signal) {
+                    $this->signals->call($signal);
+                });
+            },
+            function ($signal) {
+                if ($this->signals->count($signal) === 0) {
+                    unset($this->signalEvents[$signal]);
+                }
+            }
+        );
+    }
+
+    public function addReadStream($stream, $listener)
+    {
+        $key = (int)$stream;
+
+        if (isset($this->readStreams[$key])) {
+            return;
+        }
+
+        $callback = $this->getStreamListenerClosure($stream, $listener);
+        $event = $this->loop->io($stream, Ev::READ, $callback);
+        $this->readStreams[$key] = $event;
+    }
+
+    /**
+     * @param resource $stream
+     * @param callable $listener
+     *
+     * @return \Closure
+     */
+    private function getStreamListenerClosure($stream, callable $listener)
+    {
+        return function () use ($stream, $listener) {
+            call_user_func($listener, $stream, $this);
```

Comment:
The additional `$loop` argument has been removed via #110, so the last parameter should be omitted here.

Reply:
Thank you for pointing it! I removed this argument.